eq_cir_converter_service/services/schema/schema_processor_service.py

```diff
@@ -1,12 +1,44 @@
 """This module converts the schema from the current to the target version."""
 
+from collections.abc import Mapping, Sequence
+from typing import Any
+
+from jsonpath_ng.ext import parse
+
 from eq_cir_converter_service.config.logging_config import logging
+from eq_cir_converter_service.services.schema.extractable_strings import (
+    EXTRACTABLE_STRINGS,
+)
 from eq_cir_converter_service.types.custom_types import ConvertedSchema, InputSchema
+from eq_cir_converter_service.utils.helper_utils import process_element
 
 logger = logging.getLogger(__name__)
 
 
-async def convert_schema(current_version: str, target_version: str, schema: InputSchema) -> ConvertedSchema:
+# --- JSONPath-Based Transformation ---
+def transform_json(data: Mapping[str, Any], paths: Sequence[Mapping[str, str]]) -> Mapping[str, Any]:
+    """Transforms the JSON data based on the provided JSONPath expressions."""
+    for path_expr in paths:
+        if isinstance(path_expr, dict) and "json_path" in path_expr:
+            expr = parse(path_expr["json_path"])
+            for match in expr.find(data):
+                context = match.context.value
+                key = match.path.fields[0] if hasattr(match.path, "fields") else None
+                index = getattr(match.path, "index", None)
+
+                if isinstance(context, dict) and key:
+                    context[key] = process_element(context[key])
+                elif isinstance(context, list) and index is not None:
+                    processed = process_element(context[index])
+                    if isinstance(processed, list):
+                        context[index : index + 1] = processed
+                    else:
+                        context[index] = processed
+    return data
```

Comment:
Why do we return `data`, if `data` is not manipulated in any way? At least not explicitly, I don't see any mention of `data` object in this function other than the initial argument passing and the final `return` line. Also if we return modified `data` it should be new object copied from the `data` we pass in as argument. It is slightly confusing cause we `transform_json` but we work on and return `data` so maybe the function should change it's name to `transform_data`?